### ISSUE TITLE
Project-wide YAML metadata cannot be suppressed

### ISSUE THREAD

**Issue by mksinicus** [quarto-dev/quarto-cli#2210]
### Bug description

In a quarto website/blog project, I put this in `_quarto.yml` to set the "default" author:

```yaml
format:
  html:
    author: someAuthor
```

And I don't want to display it on the main page, so I put `author: Null` in `index.qmd`. This worked well with quarto ported with RStudio 2022.07.1 (seems to be v1.0.36), but after I installed quarto 1.1.163 in my system (Kubuntu 22.04.1), the project refuses to render, saying

```
ERROR: Validation of YAML front matter failed.
ERROR: In file index.qmd
(line 3, columns 9--12) Key author has value Null, which must be an object
2: title: "test"
3: author: Null
           ~~~
4: ---
✖ The value Null is a null value.
ℹ The error happened in location author.
```

I installed the 1.2.15 official pre-release and the problem remains. 

The problem can be minimally reproduced by 

1. Open a new quarto website project in RStudio
2. In `_quarto.yml`, set 
    ```yaml
    format:
      html:
        author: someAuthor
    ```
3. And set `author` to be `Null`, `null` or `~` in `index.qmd`. (BTW I remember having `author: null` renders as if `null` is a string with quarto 1.0.36, unsure though)

I guess it's the YAML validator that blocks the page from rendering. Since it worked fine in the earlier version, I suppose it's valid YAML...

### Checklist

- [X] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [X] Please [format your issue](https://yihui.org/issue/#please-format-your-issue-correctly) so it is easier for us to read the bug report.
- [X] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] Please document the operating system you're running. If on Linux, please provide the specific distribution.

**Comment by cscheid** [comment]
@jjallaire I can reopen that schema so this works, but my question is: was `Null` ever intended behavior in 1.0.36? I don't think I saw that documented anywhere

**Comment by mksinicus** [comment]
Yes I also tried empty string but forgot to mention. 

I then tested in R Markdown (2.15). Both `Null`, false, and empty string work for `title`, `subtitle`, and `date` etc, but fail for `author`. `Null` and empty string yield an empty `h4.author`, and `false` yields `<h4 class="author">false</h4>` (!?). I guess it's because `author` takes possible array while other fields do not. Maybe we can have them fixed and regulated in quarto.

Thank you devs!

**Comment by cscheid** [comment]
@mksinicus This is fixed on our 1.3 prerelease by using an empty string.

```yml
# _quarto.yml
      - about.qmd

format:
  html:
    author: Carlos Scheidegger
    theme: cosmo
    css: styles.css
    toc: true
```

```
---
title: "Site"
author: ""
---

This is a Quarto website.

To learn more about Quarto websites visit <https://quarto.org/docs/websites>.
```

Generated output:

<img width="1053" alt="image" src="https://user-images.githubusercontent.com/285675/214699155-de313e38-fd1d-4ae1-bdb0-ff404b543e67.png">

Without the `author: ""` line, you get this:

<img width="1053" alt="image" src="https://user-images.githubusercontent.com/285675/214699224-29462197-4aef-46d8-b7e4-13e66db050b0.png">
